tikv/pd
# statistics: get region info via core cluster inside RegionStatistics

### What problem does this PR solve?


Issue Number: Close #6560.

### What is changed and how does it work?



```commit-message
Instead of maintaining its own region info cache, use the core cluster to
fetch the region info inside `RegionStatistics` to make sure consistency.
```

### Check List



Tests



- Unit test
- Integration test
- Manual test

Run `pd-ctl region check miss-peer`:

```json
[
  {
      "id": 22,
      "start_key": "7480000000000000FF5C00000000000000F8",
      "end_key": "748000FFFFFFFFFFFFF900000000000000F8",
      "epoch": {
        "conf_ver": 1,
        "version": 56
      },
      "peers": [
        {
          "id": 23,
          "store_id": 1,
          "role_name": "Voter"
        }
      ],
      "leader": {
        "id": 23,
        "store_id": 1,
        "role_name": "Voter"
      },
      "cpu_usage": 0,
      "written_bytes": 204,
      "read_bytes": 0,
      "written_keys": 4,
      "read_keys": 0,
      "approximate_size": 1,
      "approximate_keys": 0
  }
]
```

Run `pd-ctl region 22`:

```json
{
  "id": 22,
  "start_key": "7480000000000000FF5C00000000000000F8",
  "end_key": "748000FFFFFFFFFFFFF900000000000000F8",
  "epoch": {
    "conf_ver": 1,
    "version": 56
  },
  "peers": [
    {
      "id": 23,
      "store_id": 1,
      "role_name": "Voter"
    }
  ],
  "leader": {
    "id": 23,
    "store_id": 1,
    "role_name": "Voter"
  },
  "cpu_usage": 0,
  "written_bytes": 204,
  "read_bytes": 0,
  "written_keys": 4,
  "read_keys": 0,
  "approximate_size": 1,
  "approximate_keys": 0
}
``` 

### Release note



```release-note
None.
```
